src/currentWeather/SearchHistory.tsx

```diff
@@ -0,0 +1,40 @@
+import "./SearchHistory.css";
+
+type SearchHistoryProps = {
+  searchHistory: string[];
+  onSetLocation: React.Dispatch<React.SetStateAction<string>>;
```

Comment:
Why using this react type if signature of function is basically the `(value: string) => void` ?

Reply:
Your functions in props are **events**, not react use state setters

If parent wants to set something - no problem, it's his deal
but it's not the deal of child component to declare event as react setter function